### DISCUSSION TITLE
Use of PyAbel for electron density calculation from interferogram

### DISCUSSION THREAD

**Discussion by zhangli28** [PyAbel/PyAbel#380] [Q&A]
Dear All,

I would like to know if PyAble can be used to retrieve electron density or phase map from an experimentally measured inteterferogram ?

Thank you.

**Comment by DanHickstein** [comment]
Hi @zhangli28!

Can you provide more information about what sort of data you're referring to? Maybe some example or a link to a paper that describes the experiment/technique? 

Generally speaking the Abel transform can recover a 3D object from its 2D projection. So, it's not a sophisticated retrieval procedure like that used to recover the electron density from an x-ray diffraction (crystallography) experiment, for example.

**Comment by zhangli28** [comment]
Hi @DanHickstein , 

Thank you for your quick reply. I am an experimentalist and use interferometry to check the electron density formation when an intense laser pulse incident on material. 

I would like to know if PyAbel can use this "interferogram image" and "wavelength" of the laser pulse (usually 2nd harmonic), used for making interferogram, to reconstruct the electron density profile. 

An example of using Abel inversion for electron density calculation is as follows: https://www.sciencedirect.com/science/article/pii/S0030401801015759

Thank you.

**Comment by zhangli28** [comment]
Hi @DanHickstein ,

apart from above reference, please also see fig. 2 and 3 of the attached paper. 

Thank you.

[Advanced_Diagnostics_Applied_to_a_Laser-Driven_Electron-Acceleration_Experiment.pdf]
(https://github.com/PyAbel/PyAbel/files/12721889/Advanced_Diagnostics_Applied_to_a_Laser-Driven_Electron-Acceleration_Experiment.pdf)

**Comment by MikhailRyazanov** [comment]
Let me quote the essential passages from these articles:

> Once the phase map $`\delta\phi(x, z)`$ has been extracted from the interferogram via standard FFT techniques … or with a more sophisticated wavelet-based method …

> Interferograms have been deconvolved with a fringe-analysis technique that makes use of a continuous-wavelet-transform ridge-extraction algorithm to extract the phase-difference map … These phase maps are then processed with an algorithm that generalizes the Abel inversion …

Extracting phase maps from interferograms is really unrelated to the Abel transform and is beyond the scope of PyAbel. But if you already have the phase map, and the distribution is known to be cylindrically symmetric, then PyAbel can definitely perform the inverse Abel transform of that phase map. Converting the resulting retardation distribution (or however it's called) to the electron density is again not related to the Abel transform but must be done based on whatever model describing this process.
